server/schedule/region_scatterer.go

```diff
@@ -250,18 +318,30 @@ func (r *RegionScatterer) collectAvailableStores(region *core.RegionInfo, contex
 	return targets
 }
 
-func (r *RegionScatterer) searchLeastleaderStore(peers map[uint64]*metapb.Peer, context engineContext) uint64 {
+func (r *RegionScatterer) collectAvailableLeaderStores(group string, peers map[uint64]*metapb.Peer, context engineContext) uint64 {
 	m := uint64(math.MaxUint64)
-	id := uint64(0)
+	minTableLeaderStores := map[uint64]struct{}{}
 	for storeID := range peers {
-		count := context.selectedLeader.get(storeID)
+		count := context.selectedLeader.get(storeID, group)
+		if m > count {
+			m = count
+			minTableLeaderStores = map[uint64]struct{}{}
```

Comment:
why init it twice?

Reply:
The `minTableLeaderStores` here is actually stored the candidates for the target leader store. So if we found a new "min count number" here, we should reset the `minTableLeaderStores`. Also `minTableLeaderStores`'s name is misleading, I rename it to `tableLeaderStoreCandidates`.